routes/transactionBreakdown.ts

```diff
@@ -0,0 +1,110 @@
+import { Request, Response } from 'express';
+import express from 'express';
+const router = express.Router();
+
+const createItemList = (items: Array<any>) => {
+  return items.map((individual: any, ind: number) => {
+    const indItem = {
+      id: ind,
+      name: individual.name,
+      people: individual.people,
+      cost: individual.cost,
+    };
+    return indItem;
+  });
+};
+
+const getSubAmount = (orderItems: Array<any>, name: string = '') => {
+  let total = 0;
+  for (const item of orderItems) {
+    if (name) {
+      if (item.people.includes(name)) {
+        total += Math.round((item.cost / item.people.length) * 100) / 100;
+      }
+    } else {
+      total += item.cost;
+    }
+  }
+  return total;
+};
+
+const createOrderTransaction = (
+  items: Array<any>,
+  tax: number,
+  tip: number,
+  party: Array<string>
+) => {
+  const subAmount = getSubAmount(items);
+
+  return {
+    id: 1,
+    items: items,
+    people: party,
+    amountDetail: {
+      id: 1,
+      transactionId: 3,
+      amount: subAmount + tax + tip,
+      subAmount: subAmount,
+      tax: tax,
+      tip: tip,
+    },
+  };
+};
+
+const getPersonCost = (name: string, orderList: Array<any>, orderTransaction: any) => {
+  const subAmount = getSubAmount(orderList, name),
+    costRatio = subAmount / orderTransaction.amountDetail.subAmount,
+    tax = Math.ceil(orderTransaction.amountDetail.tax * costRatio * 100) / 100,
+    tip = Math.ceil(orderTransaction.amountDetail.tip * costRatio * 100) / 100;
+
+  const totalAmount = subAmount + tax + tip;
+  return {
+    amount: totalAmount,
+    subAmount: subAmount,
+    tip: tip,
+    tax: tax,
+  };
+};
+
+const personItemList = (name: string, orderList: Array<any>) => {
+  return orderList.filter((item) => {
+    return item.people.includes(name);
```

Comment:
since the filtering is only a one liner, do we need the inner return? can't we just do:
```
orderList.filter((item) => item.people.includes(name));
```